src/io/statx.rs

```diff
@@ -0,0 +1,48 @@
+use std::{ffi::CStr, io};
+
+use io_uring::{opcode, types};
+
+use crate::runtime::{
+    driver::op::{Completable, CqeResult, Op},
+    CONTEXT,
+};
+
+use super::SharedFd;
+
+pub(crate) struct Statx {
+    fd: SharedFd,
+    statx: Box<libc::statx>,
+}
+
+impl Op<Statx> {
+    pub(crate) fn statx(fd: &SharedFd) -> io::Result<Op<Statx>> {
+        CONTEXT.with(|x| {
+            let empty_path = CStr::from_bytes_with_nul(b"\0").unwrap();
+            x.handle().expect("not in a runtime context").submit_op(
+                Statx {
+                    fd: fd.clone(),
+                    statx: unsafe { Box::new(std::mem::zeroed()) },
```

Comment:
Confused by this unsafe and the Box. Why is it needed? Should it come with a safety contract comment or am I being dense today? I'll admit to not having looked at the opcode::State::new implementation in the other crate yet.

Reply:
Okay, I'm guessing there needs to be some memory that is ready for the kernel to write into, and Box is our preferred way, allowing the future to be moved around?

Reply:
Okay. Sorry but would you mind swapping the unsafe and the Box, so it has the same structure as the places we start with a boxed msghdr? Rest looks good.

Reply:
Sure. That's no problem to change.